backend/main.py

```diff
@@ -169,7 +173,7 @@ def llm_thread(g, prompt):
                 query_text = '\n'.join(query_messages)
 
                 # add some more context
-                docs = embedding_search(query_text, k=2)
+                docs = embedding_search(query_text, k=int(os.environ['NUM_RELEVANT_FOLLOWUP_DOCS']))
```

Comment:
Why are we including `[system_message] + new_messages` in the embedding search query here? IMO this is a bug - I created issue #9 for this as well.